nav + main

annakocot opened the pull request:
zapomniałam zrobic osobny branch zanim dodałam heading...

annakocot committed:
nav

annakocot committed:
header again

annakocot committed:
main

annakocot committed:
challanges

kłopoty mam z: position: absolute, chyba szerokoscią oraz również z outline-offset, zeby zrobić nim ramke prostokatna mimo, ze obrazek jest kwadratowy

furdzik commented on index.html:
```diff
@@ -37,18 +37,19 @@
-                            <h2 class="header__heading-secondary">Find out if you can</h2>
-                            <h1 class="header__heading-primary">survive</h1>
+                        <h2 class="header__heading-secondary">Find out if you can</h2>
+                        <h1 class="header__heading-primary">survive</h1>
```
Raczej bym zrobiła h1 albo h2 eader__heading a środku na spanach ten tekst , z racji tego, że jest to jedno zdanie, BEM jest OK :)

annakocot replied:
Tu próbowałam tak zrobić jak piszesz, ale nie umiałam tego zrobić i dlatego ostataecznie podzieliłam :(

furdzik commented on index.html:
```diff
@@ -37,18 +37,19 @@
-                        <p class="header__paragraph">Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy 
+                        <p class="header__paragraph">Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
+                            nonumy
                             eirmod tempor invidunt ut labor.</p>
```
Formatowanie bym zrobiła:
< p class="header__paragraph">
     Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labor. 
</ p>
We wszystkich miejscach gdzie to występuje

annakocot replied:
done

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
+            <div class="main__posts">
+                <h4 class="main__heading">Popular challenges</h4>
+                <div class="main__post post">
+                    <div class="post__img--1">
```
"post__img--1" - to jest niepoprawny BEM, wiesz jak będzie poprawnie?

furdzik replied:
Nie podoba mi się też nazwa "post__img", ogólnie img nazwa mi się nie podoba (niżej też), bo sugeruje że to tag img (nawet jeśli tak jest to jest to mało elastyczne na potencjalne zmiany). Można wykorzystać takie nazwy jak: image, photo, cover-photo itp.

furdzik replied:
Podciągnęłabym tego diva początek i zakończenie w tej samej linii (we wszystkich miejscach gdzie to występuje)

annakocot replied:
Naprawiłam BEM, zmieniłam na image i przesunełam diva do jednej linii.

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
+                <div class="main__post post">
+                    <div class="post__img--1">
+                    </div>
+                    <div class="post__textbox">
```
Może "post__description"?

furdzik replied:
Dodatkowo jeśli chodzi o BEMa, to duży ten blog, zastanów się czy nie wydzielić mniejszych w środku tego jednego dużego, ładniej kod htmlowy ale też cssowy będzie wyglądać i będzie bardziej czytelny

annakocot replied:
Nie wiem za bardzo jak to zrobić, żeby miało ręce i nogi.

furdzik replied:
To spróbujemy razem w weekend

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
+                    <div class="post__textbox">
+                        <h6 class="post__subheading">key skills</h6>
+                        <h5 class="post__heading">Friction-Based Fire Making</h5>
+                        <p class="post__paragraph">Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse
```
Tu też formatowanie jakbyś mogła poprawić

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
                             consequatur, vel illum qui dolorem.</p>
-                        <a class="more" href="">find out more</a>
+                        <div class="post__morebox">
+                            <a class="post__more" href="">find out more</a>
```
Może lepiej "post__more-link"? Co uważasz?

annakocot replied:
tak zrobiłam

furdzik commented on style/layout/_nav.scss:
```diff
@@ -6,10 +6,12 @@
+            width: 90%;
+        }
 
+    &__nav{
```
Brakuje spacji

furdzik commented on style/_all.scss:
```diff
@@ -3,12 +3,13 @@
-@import 'layout/nav';
\ No newline at end of file
+@import 'layout/nav';
+@import 'layout/main';
```
Dobrą praktyką jest zostawianie jednej puste linii na końcu każdego pliku. Więcej na ten temat:
https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file
https://thoughtbot.com/blog/no-newline-at-end-of-file
Można to uzyskać automatycznie, w tym 2 artykule masz informacje jak to skonfigurować (chyba, bo nie wiem jak to wygląda w VS Code, mam nadzieje że Ty zrozumiesz :P)

annakocot replied:
OK udało mi się to ustawić :)

furdzik commented on style/base/_typography.scss:
```diff
@@ -28,10 +28,10 @@ h5{
+// h6{
+//     font-family: $primary-font;
+//     font-size: $font-size-small;
+// }
```
Rozumiem, że ten komentarz chwilowo. Przejrzyj pliki pod kątem formatowania, bo w wielu miejscach brakuje ci spacji przed "{"

annakocot replied:
no nie wiem czy w ogóle tego typography nie wyrzucić w pierony.....

furdzik replied:
Tylko może Ci się rozwialić, chyba że i tak to nadpisywałaś. To jak komu wygodnie. Często się robi w projektach taką normalizację (zamiast resetu). Jeszcze o tym wspomnę, wyśle jakiegoś linka :)

annakocot replied:
No, ja właśnie gdzieś indziej tak widziałam i dlatego tak samo chciałam zrobić a okazuje się, że i tak nadpisałam wszystko, co używam.

furdzik commented on style/layout/_challenges.scss:
```diff
@@ -0,0 +1,126 @@
+        text-transform: uppercase;
+        margin-left: 4rem;
+
+        &::before{
```
Brakuje spacji

furdzik commented on style/layout/_challenges.scss:
```diff
@@ -0,0 +1,126 @@
+            font-size: $font-size-big;
+            color: $color-primary;
+            display: inline-block;
+            transform: rotate(-45deg);
```
Tego trochę nie rozumiem. Po co przekręcasz tego plusa?

annakocot replied:
w layoucie to jest X, ale taki nie jak X tylko przekręcony plus właśnie

furdzik replied:
A to nie lepiej po prostu "x"

annakocot replied:
właśnie "x" ma pod innym kątem linie niż +

furdzik replied:
To ikonką się takie rzeczy robi :)

annakocot replied:
zrobiłam ikonką i zepsuło sie wszystko :(

furdzik commented on style/layout/_header.scss:
```diff
@@ -64,26 +64,22 @@
 
-        ul {
-            list-style-type: none;
+    &__social-list    {
```
Za dużo spacji i złe wcięcia między 69 a 75 linijką

furdzik commented on style/layout/_main.scss:
```diff
@@ -0,0 +1,197 @@
+        margin: 0 2rem 2rem 2rem;
+
+        &--1 {
+            background: url('../../img/fire1.svg');
```
Jak poprawkisz BEMa, to te powtarzające się style będą w głownej deklaracji, więc w tych --1 --2 będziesz miała tylko background.
Dodatkowo, jak piszesz background z obrazkiem w środku, to określaj też pozycję i powtarzanie czyli: background: url(...) no-repeat; a jak potrzebujesz na środku to: background: url(...) center center no-repeat;

furdzik replied:
Pamiętaj też, że jeśli możesz jakoś uniknąć powtarzania tych samych reguł w wielu klasach. Np poprzez przecinek w selektorach (.clasa-1, . clasa-2 {....}) albo nadanie jednej wspólnej klasy dodatkowej albo nadrzędnej. Jak poprawisz tego BEMa to myśle, że Ci się to rozjaśni, jakby co to pytaj :)

annakocot replied:
Właśnie miałam troche powtarzających się rzeczy, ale w różnych elementach np. header__paragraph i challenge__paragraph i nie wiedziałam jak je sobie wyciągnąć, ale teraz myślę, że może mogę zrobić mixin i sobie go potem do obu include?

furdzik replied:
Jeśli zmieniają się zmienne jakieś to może być mixin ale jeśli są takie same style to zwykły extend będzie lepszy.

furdzik commented on style/layout/_main.scss:
```diff
@@ -0,0 +1,197 @@
+            font-size: $font-size-big;
+            color: $color-primary;
+            display: inline-block;
+            transform: rotate(-45deg);
```
Powtarzasz te style kilka razy, w różnych blokach, możesz stworzyć klasę ogólną i ją tutaj excludować. Są dwa sposoby na to aby to robić. Tu masz info: https://sass-lang.com/documentation/at-rules/extend 
Jak jeszcze gdzieś masz coś takiego, co ja nie zauważyłam to też można zastosować oczywiście, jak masz wątpliwości to pytaj :)

annakocot replied:
zrobiłam mixin, bo trochę nie rozumiem tego extend...

furdzik replied:
To nie dobrze, bo jeśli nie zmieniają się żadne style to mixin jest zbędny lepiej użyć extenda, mozemy to zrobić razem, nauczę Cię tego :)

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
+                </h2>
+                <h3 class="mission__back-heading">
+                    challenges
+                </h3>
```
Nie pamiętaj projektu, ale wydaje mi się, że to też powinno być w jednyk h i spanem rozdizelone style

annakocot replied:
tu ten h3 jest na absolute i jest na wierzchu h2 i tak po skosie, więc nie wiem - da się to zrobić w span?

furdzik replied:
Wszystko się da, ale wyślij mi proszę projekt to jeszcze się zastanowie :P albo razem mozemy się zastanowić

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
+                    <p class="mission__paragraph">
+                        Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam
+                        nihil molestiae consequatur, vel illum qui dolorem.</p>
+                    <a class="mission__more" href="">find out more</a>
```
Wygląda jak złe formatowanie na pierwszy rzut oka, ja bym </ p> dała do niższej linijki :)
Odnośnie formatowania jeszcze możemy pogadać, poszukam Ci jakiś linków, choć może to być trudne, ja to sobie sama wypracowałam przez lata.

furdzik commented on index.html:
```diff
@@ -67,50 +68,107 @@ <h1 class="header__heading-primary">survive</h1>
+                <h2 class="challenges__results">Your restults</h2>
+                <h6 class="challenges__completed">
+                completed challenges
+            </h6>
```
Złe wcięcia

furdzik commented on style/layout/_challenges.scss:
```diff
@@ -0,0 +1,126 @@
+    }
+
+    &__photo {
+        background: url('../../img/img\ man.svg');
```
Coś tu nie tak z "/"

furdzik commented on style/layout/_challenges.scss:
```diff
@@ -0,0 +1,126 @@
+        position: absolute;
+        top: 30rem;
+        left: 22rem;
+
```
Niepotrzebny enter

furdzik commented on style/layout/_challenges.scss:
```diff
@@ -0,0 +1,126 @@
+        font-size: $font-size-big;
+        color: $color-white;
+        text-transform: uppercase;
+       max-width: 35rem;
```
Złe wcięcia, niżej też

furdzik commented on style/layout/_challenges.scss:
```diff
@@ -0,0 +1,126 @@
+        text-transform: uppercase;
+        margin-left: 4rem;
+
+        &::before{
```
Brak spacji

furdzik commented on style/layout/_main.scss:
```diff
@@ -0,0 +1,197 @@
+        font-weight: 500;
+        text-transform: uppercase;
+        word-break: break-all;
+        line-height: 0.7;
```
Rozmiar czcionki masz xxl a line-height: 0.7. Mam pewne wątpliwości czy do dobry pomysł, nie pamiętam wyglądu, ale możliwe, że jest lepszy sposób na osiągnięcie efektu, który próbujesz uzyskać. Przegadajmy to jeszcze na żywo.

furdzik commented on style/layout/_main.scss:
```diff
@@ -0,0 +1,197 @@
+        max-width: 40rem;
+
+        &::before {
+            content: "01";
```
Znów, nie pamiętam layoutu, ale nie widzę nigdzie 02, 03? To na prwno jest numerowanie czy może tekst (który np może być inny). Wyobrażaj sobie, że tekst który masz w środku może być przykładowy, albo generyczny, to znaczy zmieniać się jak podepnie się np. szablon pod CMSa. Oczywiście są projekty, które trzeba w 100% odwzorować zgodnie z projektem, musisz też wiedzieć do jakiego przeznaczenia jest dana htmlka. Też możemy o tym więcej porozmawiać, bo mogłaś tego dokładnie nie zrozumieć, ciężko to tak wyjaśnić, ale takie rzeczy są dość istotne przy tworzeniu kodu :)

annakocot replied:
Nie ma nigdzie 02, 03. Nie wiem do końca czemu to służy. Musisz mi powiedzieć, bo nie wiem czy Cie rozumiem...

furdzik replied:
To to też obgadajmy na spotkaniu

furdzik commented on style/layout/_main.scss:
```diff
@@ -0,0 +1,197 @@
+        margin-bottom: 2rem;
+    }
+
+    &__paragraph {
```
Szczerze to ta nazwa "paragraph" też mi się nie podoba, bo znów kolaży się z tagiem "p", lepiej używać nazw klas mówiących co to jest, aniżeli tagi, czy wygląd.

annakocot replied:
Ok, to zamienie to na coś innego tylko nie wiem na co za bardzo...

furdzik replied:
"description" np?

furdzik reviewed (changes requested):
Generalnie naprawdę bardzo ładnie sobie radzisz. Ilością komentarzy się nie przejmuj, na początku zawsze jest ich więcej, dzięki temu się uczysz :)

Tematy które fajnie by było  żebyś przemyślała i o nich pogadamy, opowiem Ci jakie są trendy, jak ja robię itp:
- Formatowanie CSS i HTML
- Kolejność reguł CSSowych w każdej deklaracji CSS (reguła to: np display: block;, deklaracja to łącznie z nazwą klasy itp.)

Nazwa pull requestów, gałęzi, komitów:
- Też potrzebnt jest jakiś styl guide na któym można się oprzeć, to też mogę Ci coś zaproponować i czy pokazać jak ja to robię
- Przyjmuje się że nazwy commitiów, pullreqestów, gałęzi pisze się po angielsku
- Nazwy komitów powinny obrazować o co chodzi w danym kominie, mówi się, że lepiej robić więcej komitów z konkretnymi zmianami i dobrze to nazwać, podeśle Ci przykłady i poszukam linków gdzie będzie opisane to, na pewno Ci się przyda w przyszłej pracy :)

Dodatkowo:
Są takie rzeczy, których w code review nie idzie wyłapać, więc możemy jeszcze razem przejść przez kod i zweryfikować razem :)

furdzik commented:
Swoją drogą wykonywanie code review to też można o tym pogadać, bo na pewno będziesz robić. To zastanów się, jak chcesz to poczytaj odnośnie tego tematu, co jest ważne itp. i pogadamy o tym też :)

annakocot committed:
fixed&improved

annakocot committed:
mixins created

annakocot committed:
advice section

annakocot committed:
advice pages and arrows

annakocot committed:
font added succesfully

annakocot committed:
gallery section added

annakocot committed:
footer added

annakocot commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,52 @@
+.footer {
+    background: url('../../img/img_footer.svg') no-repeat;
```
Tu mam problem jak ustawić pozycje obrazka tła, większego niż obszar tła. Chciałam zrobić to background-position: center, ale wtedy też nie jest jak w projekcie no i nie wiem.

furdzik replied:
Możemy to też obgadać na żywo, nie nie do końca czaje o co Ci chodzi :D

annakocot committed:
cleaned and improved

zrobiłam z main main i w nim mniejsze sekcje, dla more utworzylam osobna klase, zeby wszedzie bylo tak samo, poprawiłam se zdjęcie w footer, w końcu ogarnęłam jak je przesunąć i tak ogólnie troche posprzątałam

annakocot committed:
tabs reduced to 2 spaces

annakocot committed:
code fixed, animationd added

annakocot committed:
rwd footer

furdzik commented on style/base/_typography.scss:
```diff
@@ -1,39 +1,61 @@
+
+ul,
+ol {
+  list-style: none;
```
typografpgy oznacza czcionki a zrobiłaś tu taki reset / normalize / basic css troche, w typography zostaw same przypisanie czcionek + te nagłowki i p -> kolory, font-weight może zostać. a i ol, ul przenieś gdzieś indziej

furdzik commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,220 @@
+
+  @include media-breakpoint-up(landscapePhone) {
+    height: 56rem;
+    width: 56rem;
```
najpierw width potem height

furdzik commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,220 @@
+    @include media-breakpoint-up(fhdDesktop) {
+      width: 28rem;
+      height: 14rem;
+    }
```
wydaje mi się, że niepowinno być koniecznie zmiana rozmiaru loga per brakepoint, ale cięzko mi jest to sobie wyobrazić, to może jutro jeszcze to obgadamy. Jak będziesz chciała spróbuj wykorzystać max-with, min-with nie zawsze musisz dawać szerokości na sztywno

furdzik commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,220 @@
+    @include media-breakpoint-up(fhdDesktop) {
+      margin-left: 29rem;
+      max-width: 80rem;
+    }
```
Tu tak samo, te wartości z poszczególnych brakepointów są dziedziczone (jak masz mobile first to te co są niżej dziedziczą te niższe wartości, pozatym zmiana z 30 na 32, troche to dziwne. Jeszcze to zobacze czy to jest dobrze przemyślane, często z RWD zmiania się sposób pisania stylów)

furdzik commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,220 @@
+
+    @include media-breakpoint-up(fhdDesktop) {
+      left: 34rem;
+    }
```
znów cała litania media quesy, możliwe że niepotrzebna

furdzik commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,220 @@
+    text-align: center;
+
+    @include media-breakpoint-up(desktop) {
+      flex-direction: row;
```
tego typu zmiany - super

furdzik commented on style/layout/_footer.scss:
```diff
@@ -0,0 +1,220 @@
+    padding: 2rem;
+
+    @include media-breakpoint-up(landscapePhone) {
+      font-size: 5rem;
```
powtarzasz to samo

furdzik commented on style/layout/_header.scss:
```diff
@@ -1,143 +1,91 @@
-        padding: 1rem;
-        background: #fff;
-        cursor: pointer;
+  &__item+&__item {
```
&__item + &__item - spacje przed i po + :)
